### ISSUE TITLE
Fuzzy logic allowing overcoming rule overlaps/clashes/ambiguities/...

### ISSUE THREAD

**Issue by dumblob** [samsquire/additive-guis#5]
I know this is just a prototype demo, but I still think it should be closer to what people mean when they say `leftOf` `rightOf` etc. People do not think in terms of the whole system (i.e. they don't derive all the consequences of their previous rules) and thus there are ambiguities, imprecise intents (often also deliberately!), overlaps, clashes etc.

How about adding some fuzzy "weighting" to rules and use some simple fuzzy logic algo to get the final "sharp" value? Default for `leftOf` would be actually `rather leftOf` but one could also say `fully leftOf` or `slightly leftOf`.

In other words, all the rules would be by default `rather` to give the solver algorithm more space to solve the rules while at the same offering the user with a simple way of providing a "weight" (importance, emphasis) of the given rule.

Thoughts?

**Comment by samsquire** [comment]
Yes I would like to implement this.

I have been reading the ORCSolcer algorithm and it handles layouts of inequalities in a different more proper way. My inequality handling is trivially simple.

I want to use a proper solver.

https://github.com/YueJiang-nj/ORCSolver-CHI2020

**Comment by dumblob** [comment]
Yes, I have seen that paper some time ago. It is interesting but actually still quite slow. But that is not a reason to not use it (besides it seems to be state of the art in that category).

Still the ORC solver is not "fuzzy" from what I remember. But maybe it is possible to add fuziness on top of it. IDK. It definitely needs experimentation :wink:.
